### PR TITLE
Repair - Fix bug in detection for ignoring hitpoints

### PR DESCRIPTION
**Pull request by tbeswick96**
**When merged this pull request will:**
- Fix bug in detection for ignoring repair hitpoints

### CODE REVIEW

File: addons/repair/functions/fnc_getHitPointsToIgnore.sqf [hunk 1]
```diff
@@ -123,7 +123,7 @@ private _processedSelections = [];
         continue
     };
 
-    if ((_hitpointGroups findIf {(_x select 1) == _hitpoint}) != -1) then { // skip child hitpoints
+    if ((_hitpointGroups findIf {((_x select 1) findIf {_x == _hitpoint}) != -1}) != -1) then { // skip child hitpoints
```
Comment (BaerMitUmlaut) [inline addons/repair/functions/fnc_getHitPointsToIgnore.sqf:126]: At this point this should be split into multiple lines. This is getting seriously unreadable 😅

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

File: addons/repair/functions/fnc_getHitPointsToIgnore.sqf (hunk 1 above)
Comment (commy2) [reply addons/repair/functions/fnc_getHitPointsToIgnore.sqf:126]:
```sqf
#define ANY(set,condition) (set findIf {condition} != -1)

if ANY(_hitpointGroups, ANY(_x select 1, _x == _hitpoint))
```

**Review by LinkIsGrim — APPROVED** [review]

**Comment by PabstMirror** [comment]
my changes to this could use another eyes
```
TRACE: 6396 Skipping child hitpoint: _hitpoint=hitengine1, _forEachIndex=2, _selection=engine_1_hit z\ace\addons\repair\functions\fnc_getHitPointsToIgnore.sqf:134
TRACE: 6396 Skipping child hitpoint: _hitpoint=hitengine2, _forEachIndex=3, _selection=engine_2_hit z\ace\addons\repair\functions\fnc_getHitPointsToIgnore.sqf:134
TRACE: 6396 Skipping depends hitpoint: _hitpoint=hitengine, _forEachIndex=4, _selection=engine_hit z\ace\addons\repair\functions\fnc_getHitPointsToIgnore.sqf:117
ERROR: [O_Heli_Transport_04_F] hitpoint [hitengine] is both a group-parent and a depends and will be unrepairable
ERROR: group: ["HitEngine",["HitEngine1","HitEngine2"]]
```

there was another bug with this, the entire hitpoint group became unrepairable because
we ignored the children
but the parent was also a "depends" hitpoint and was also ignored

I think the reason for groups was long before the depends system and may not be needed anymore
this removes the one use of hitpointGroups config, I'm not sure of other addons usage but we can keep an eye out

**Review by PabstMirror — APPROVED** [review]

File: addons/repair/CfgVehicles.hpp [hunk 2]
```diff
@@ -438,9 +438,7 @@ class CfgVehicles {
     };
 
     class Helicopter_Base_H;
-    class Heli_Transport_04_base_F: Helicopter_Base_H {
-        GVAR(hitpointGroups)[] = { {"HitEngine", {"HitEngine1", "HitEngine2"}} };
-    };
+    class Heli_Transport_04_base_F: Helicopter_Base_H {};
```
Comment (LinkIsGrim) [inline addons/repair/CfgVehicles.hpp:441]:
```suggestion
    class Heli_Transport_04_base_F;
```

File: addons/repair/CfgVehicles.hpp [hunk 3]
```diff
@@ -438,9 +438,7 @@ class CfgVehicles {
     };
 
     class Helicopter_Base_H;
```
Comment (LinkIsGrim) [inline addons/repair/CfgVehicles.hpp:440]:
```suggestion
```

**Review by LinkIsGrim — COMMENTED** [review]
Does this break anything?

File: addons/main/script_macros.hpp [hunk 4]
```diff
@@ -38,6 +38,7 @@
 #define GETEGVAR(var1,var2,var3) GETMVAR(EGVAR(var1,var2),var3)
 
 #define ARR_SELECT(ARRAY,INDEX,DEFAULT) (if (count ARRAY > INDEX) then {ARRAY select INDEX} else {DEFAULT})
+#define ANY(ARRAY,CONDITION) (ARRAY findIf {CONDITION} != -1)
```
Comment (PabstMirror) [inline addons/main/script_macros.hpp:41]:
```suggestion
#define ANY_OF(ARRAY,CONDITION) (ARRAY findIf {CONDITION} != -1)
```

File: addons/repair/functions/fnc_getHitPointsToIgnore.sqf [hunk 5]
```diff
@@ -118,12 +118,19 @@ private _processedSelections = [];
...
+            ERROR_1("group: %1",_hitpointGroups # _groupIndex);
+        };
+
         _hitPointsToIgnore pushBackUnique _hitpoint;
         _processedSelections pushBack _selection;
         continue
     };
 
-    if ((_hitpointGroups findIf {(_x select 1) == _hitpoint}) != -1) then { // skip child hitpoints
+    if (ANY(_hitpointGroups, ANY(_x select 1, _x == _hitpoint))) then { // skip child hitpoints
```
Comment (PabstMirror) [inline addons/repair/functions/fnc_getHitPointsToIgnore.sqf:133]:
```suggestion
    if (ANY_OF(_hitpointGroups, ANY_OF(_x select 1, _x == _hitpoint))) then { // skip child hitpoints
```

**Comment by tbeswick96** [comment]
Bad merge, will repair

**Pull request merged**